app/scripts/directives/multiSelectMenu.js

```diff
@@ -58,17 +58,17 @@ angular.module('gsUiInfraApp')
                 }
 
                 /**
-                 * Bind data to ng-model
+                 * call onchange function when model change
                  */
-                $scope.$watch('selected', function (newValue) {
-                    if (newValue) {
-
-                        setValue(newValue);
-                        if(angular.isFunction($scope.onchange)) {
-                            $scope.onchange({filter: newValue});
+                if(angular.isFunction($scope.onchange)) {
```

Comment:
besides, I think angular already has this: 
https://docs.angularjs.org/api/ng/directive/ngChange

my comment was about the `if` statement, but it doesn't matter since ng-change solves this.